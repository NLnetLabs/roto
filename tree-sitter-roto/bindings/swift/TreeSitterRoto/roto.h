#ifndef TREE_SITTER_ROTO_H_
#define TREE_SITTER_ROTO_H_

typedef struct TSLanguage TSLanguage;

#ifdef __cplusplus
extern "C" {
#endif

const TSLanguage *tree_sitter_roto(void);

#ifdef __cplusplus
}
#endif

#endif // TREE_SITTER_ROTO_H_
