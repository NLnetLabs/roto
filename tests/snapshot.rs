use insta::{assert_snapshot, glob};
use roto::{FileTree, Runtime};

#[test]
fn parse_errors() {
    let root = std::env::current_dir().unwrap();
    glob!("scripts/parse_errors/", "*.roto", |path| {
        let runtime = Runtime::new();

        let relative_path = path.strip_prefix(&root).unwrap();
        let file_tree = FileTree::read(relative_path);
        let res = file_tree.compile(runtime);
        let Err(e) = res else {
            panic!("should not succeed");
        };
        let mut string = String::new();
        e.write(&mut string, false).unwrap();
        assert_snapshot!(string);
    });
}

#[test]
fn type_errors() {
    let root = std::env::current_dir().unwrap();
    glob!("scripts/type_errors/", "*.roto", |path| {
        let runtime = Runtime::new();

        let relative_path = path.strip_prefix(&root).unwrap();
        let file_tree = FileTree::read(relative_path);
        let res = file_tree.compile(runtime);
        let Err(e) = res else {
            panic!("should not succeed");
        };
        let mut string = String::new();
        e.write(&mut string, false).unwrap();
        assert_snapshot!(string);
    });
}
