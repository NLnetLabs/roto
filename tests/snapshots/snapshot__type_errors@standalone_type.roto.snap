---
source: tests/snapshot.rs
expression: string
input_file: tests/scripts/type_errors/standalone_type.roto
---
Error: Type error: expected a value, but found type `String`
   ╭─[ tests/scripts/type_errors/standalone_type.roto:2:10 ]
   │
 1 │ filter foo() {
   │ ┬  
   │ ╰── type `String` defined here
 2 │     let a = String;
   │             ───┬──  
   │                ╰──── expected a value
───╯
